protocol/clproto_cpp/test/tests/test_parameter_proto.cpp

```diff
@@ -6,6 +6,43 @@
 
 using namespace state_representation;
 
+TEST(ParameterProtoTest, EncodeDecodeParameterInt) {
+  int value = 1;
+
+  auto send_state = Parameter("A", value);
+  std::string msg = clproto::encode(send_state);
+  EXPECT_TRUE(clproto::is_valid(msg));
+  EXPECT_TRUE(clproto::check_message_type(msg) == clproto::MessageType::PARAMETER_MESSAGE);
+  EXPECT_TRUE(clproto::check_parameter_message_type(msg) == clproto::ParameterMessageType::INT);
+
+  Parameter<int> recv_state("");
+  EXPECT_NO_THROW(clproto::decode<Parameter<int>>(msg));
+  EXPECT_TRUE(clproto::decode(msg, recv_state));
+
+  EXPECT_STREQ(send_state.get_name().c_str(), recv_state.get_name().c_str());
+  EXPECT_EQ(send_state.get_value(), recv_state.get_value());
```

Comment:
Because of the comment about StateType mapping, the following test would probably fail:
```c++
EXPECT_EQ(send_state.get_type(), recv_state.get_type())`
```

Could you add this test coverage in each parameter variant?